.githooks/pre-commit

```diff
@@ -1,34 +1,44 @@
 #!/bin/sh
 
-# If any command fails, exit immediately with that command's exit status
-set -eo pipefail
+function check_flake8() {
+  changes="$( \
+    git status --porcelain \
+      | awk '{ print $2 }' \
+      | grep '\.py$' || printf '' \
+  )"
+  if [[ -z "$changes" ]]; then
+    echo 'No python changes detected, skipping pre-commit flake8'
+    return 0  # Skip flake8 because we have no changed python code.
+  fi
+
+  # Run flake8 against all code in the `source_code` directory
+  # Only output failures.
+  echo 'Running pre-commit flake8 (run again with --no-verify to skip)...'
 
-changes="$( \
-  git status --porcelain \
-    | awk '{ print $2 }' \
-    | grep '\.py$' || printf '' \
-)"
-if [[ -z "$changes" ]]; then
-  echo 'No python changes detected, skipping pre-commit flake8'
-  exit 0  # Skip flake8 because we have no changed python code.
-fi
+  unstaged="$( \
+    git ls-files -m \
+      | grep '\.py$' \
+      | awk '{ print "- " $0 }' || printf '' \
+  )"
+  if [[ -n "$unstaged" ]]; then
+    echo
+    echo 'Warning: your unstaged changes may affect flake8 results:'
+    echo "$unstaged"
+    echo
+  fi
 
-# Run flake8 against all code in the `source_code` directory
-# Only output failures.
-echo 'Running pre-commit flake8 (run again with --no-verify to skip)...'
+  docker --log-level ERROR compose run --interactive --no-TTY web flake8
+  return $?
+}
 
-unstaged="$( \
-  git ls-files -m \
-    | grep '\.py$' \
-    | awk '{ print "- " $0 }' || printf '' \
-)"
-if [[ -n "$unstaged" ]]; then
-  echo
-  echo 'Warning: your unstaged changes may affect flake8 results:'
-  echo "$unstaged"
-  echo
-fi
+function check_prettier() {
+  echo 'Running pre-commit prettier (run again with --no-verify to skip)...'
+  npm run lint:check
```

Comment:
If the check returns linting errors, could we allow users to run `lint:write` ? Also, will reported errors prevent the commit from finishing?

Reply:
Reported errors _will_ prevent the commit from finishing (intended behavior - we don't want to commit broken things. Adding --no-verify will let us commit anyway).

I wouldn't want to modify the code in this script (i.e. run lint:write) because of the variety of ways you might have code staged, unstaged, etc - so users are unfortunately on their own to `lint:write`

Reply:
Excellent. Thanks @okkays 

Great work as usual!